File: BikeRental/BikeRentalServer/src/main/java/com/example/BikeRental/Bike/Bike.java
package com.example.BikeRental.Bike;

import com.example.BikeRental.Manufacturer.Manufacturer;
import com.example.BikeRental.User.User;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.Date;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Bike {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer bike_id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "manufacturer_id")
    private Manufacturer manufacturer;
    private String model;
    private Integer year;
    private Float price;
    private Enum<BikeStatus> status;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User currentRenter;
    private Date rentDate;

    private enum BikeStatus{
        Available,
        Rented,
        Maintenance
    }
}

File: BikeRental/BikeRentalClient/BikeRentalClient/Bike.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace BikeRentalClient
{
    internal class Bike
    {
        public int id { get; set; } = 0;
        public Manufacturer manufacturer { get; set; } = new Manufacturer();
        public string model { get; set; } = string.Empty;

        public int year { get; set; } = 0;
        public List<Component> components { get; set; } = new List<Component>();
        public float price { get; set; } = 0.0f;
        public BikeStatus status {  get; set; }
        [JsonConverter(typeof(JsonStringEnumConverter))]

        internal enum BikeStatus
        {
            Available,
            Rented,
            Maintenance
        }
    }
}
